### PR TITLE
Parameterize hardcoded strings in pdf generation

### LINKED ISSUE #661: Parameterize PDF hardcoded strings

### PR DESCRIPTION
**Pull request by Isti01**
closes #661

### CODE REVIEW

File: backend/src/main/kotlin/hu/bme/sch/cmsch/component/token/TokenAdminTokensByUsersOfGroupsController.kt [hunk 1]
```diff
@@ -116,27 +116,26 @@ class TokenAdminTokensByUsersOfGroupsController(
...
-            ClassPathResource("/static/images/sssl-logo.png").inputStream.readAllBytes()
-        )).scaleToFit(70f, 70f)
-
         val font = PdfFontFactory.createFont("OpenSans-Regular.ttf")
         val header = Paragraph()
-            .add(ssslLogo)
-            // FIXME: place it into a setting
-            .add(Paragraph("GÓLYAKÖRTE 2023\nJELENLÉTI - ${group.name}")
+        readAsset(tokenComponent.reportLogo.getValue().replace("/cdn/", "/")).map {
+            Image(ImageDataFactory.create(it,)).scaleToFit(70f, 70f)
```
Comment (Gerviba) [inline backend/src/main/kotlin/hu/bme/sch/cmsch/component/token/TokenAdminTokensByUsersOfGroupsController.kt:122]: The comma is not required here after the `it`

File: backend/src/main/kotlin/hu/bme/sch/cmsch/component/token/TokenAdminTokensByUsersOfGroupsController.kt [hunk 2]
```diff
@@ -116,27 +116,26 @@ class TokenAdminTokensByUsersOfGroupsController(
...
+        }.ifPresent(header::add)
+
+        val eventName = tokenComponent.reportTitle.getValue()
+        header.add(Paragraph("${eventName}\nJELENLÉTI - ${group.name}")
```
Comment (Gerviba) [inline backend/src/main/kotlin/hu/bme/sch/cmsch/component/token/TokenAdminTokensByUsersOfGroupsController.kt:126]: I might use a title variable here and not hardcode the 'JELENLÉTI' string.

File: backend/src/main/kotlin/hu/bme/sch/cmsch/component/token/TokenAdminTokensByUsersOfGroupsController.kt [hunk 3]
```diff
@@ -116,27 +116,26 @@ class TokenAdminTokensByUsersOfGroupsController(
...
                 .setTextAlignment(TextAlignment.CENTER)
                 .setVerticalAlignment(VerticalAlignment.MIDDLE)
                 .setFont(font)
                 .setFontSize(28f)
                 .setMarginLeft(40f)
                 .setMarginRight(40f))
-            .add(kirdevLogo)
-            .setTextAlignment(TextAlignment.CENTER)
+
+        readAsset("/static/images/kirdev-logo.png").map {
```
Comment (Gerviba) [inline backend/src/main/kotlin/hu/bme/sch/cmsch/component/token/TokenAdminTokensByUsersOfGroupsController.kt:134]: Maybe make it optional using a boolean config variable? I'm not really sure.

File: backend/src/main/kotlin/hu/bme/sch/cmsch/component/token/TokenComponent.kt [hunk 4]
```diff
@@ -123,4 +128,28 @@ class TokenComponent(
...
+
+    final val reportTitle = SettingProxy(componentSettingService, component,
+        "reportTitle", "GÓLYAKÖRTE 2025",
+        fieldName = "Jelenléti ív címe", description = "Ez lesz a jelenléti ív címe"
+    )
+
+    final val reportLogo = SettingProxy(componentSettingService, component,
+        "reportLogo", "/cdn/manifest/report-logo.png",
+        fieldName = "Jelenléti ív logója", description = "Ez lesz a jelenléti íven megjelenített logó (csak PNG és JPG jó)",
+        type = SettingType.IMAGE,
```
Comment (Gerviba) [inline backend/src/main/kotlin/hu/bme/sch/cmsch/component/token/TokenComponent.kt:147]: Do we have that setting type? I didn't remember that! :D

File: backend/src/main/kotlin/hu/bme/sch/cmsch/util/Utility.kt [hunk 5]
```diff
@@ -29,15 +32,17 @@ class DI(
     }
 }
 
+private fun getFileStoragePath(): String = if (!DI.instance.startupPropertyConfig.external.startsWith("/")) {
+    System.getProperty("user.dir") + "/" + DI.instance.startupPropertyConfig.external
```
Comment (Gerviba) [inline backend/src/main/kotlin/hu/bme/sch/cmsch/util/Utility.kt:36]: I'd just use the working directory for root and not the `user.home`

**Review by Gerviba — APPROVED** [review]

File: backend/src/main/kotlin/hu/bme/sch/cmsch/component/token/TokenAdminTokensByUsersOfGroupsController.kt (hunk 3 above)
Comment (Isti01) [reply backend/src/main/kotlin/hu/bme/sch/cmsch/component/token/TokenAdminTokensByUsersOfGroupsController.kt:134]: No no, we must show the kir-dev logo at every opportunity

File: backend/src/main/kotlin/hu/bme/sch/cmsch/component/token/TokenComponent.kt (hunk 4 above)
Comment (Isti01) [reply backend/src/main/kotlin/hu/bme/sch/cmsch/component/token/TokenComponent.kt:147]: Yes, it's for setting manifest images, and thats why I set it like /cdn/manifest/...

File: backend/src/main/kotlin/hu/bme/sch/cmsch/util/Utility.kt (hunk 5 above)
Comment (Isti01) [reply backend/src/main/kotlin/hu/bme/sch/cmsch/util/Utility.kt:36]: I moved this code from the file upload function, I don't exactly understand why it works like this, so I didn't want to change it

File: backend/src/main/kotlin/hu/bme/sch/cmsch/component/token/TokenAdminTokensByUsersOfGroupsController.kt (hunk 1 above)
Comment (Isti01) [reply backend/src/main/kotlin/hu/bme/sch/cmsch/component/token/TokenAdminTokensByUsersOfGroupsController.kt]: removed

**Pull request merged**